ouroboros-consensus/docs/AbstractProtocol.md

```diff
@@ -0,0 +1,76 @@
+# Abstract Protocol
+
+The following diagram depicts the relation between various constructs in the
+abstract specification of the protocol and the consensus-level view of th
+ledger.
+- Red boxes indicate concepts, or informal kinds (e.g., `ShelleyLedgerState`
+  would have the informal kind `Ledger`).
+- Blue boxes indicate data families.
+- Green boxes indicate type families.
+- Type or data families map the block attached to the incoming arrow to the
+  block attached to the outgoing arrow. When there is no outgoing arrow, the
+  family maps to any type (e.g., `BlockConfig blk`).
+
+```mermaid
+flowchart LR;
+
+  classDef df fill:#4285F4, border:10px;
+  classDef tf fill:#34A853;
+  classDef kind fill:#EA4335;
+
+  Block(Block)-->BlockProtocol[BlockProtocol blk]; class Block kind;
+  click Block "https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-consensus/src/Ouroboros/Consensus/Block/Abstract.hs"
```

Comment:
One thing that I worry about is that the links might break if we move code around. One way we can prevent this is by adding a CI check. But that's outside the scope of this issue. Should we create a new one for this?

Reply:
I agree. Not sure what the best solution is. I dislike having to change things in unexpected places, and I also dislike the slowness of CI checks. I need to give it some thought.

Reply:
We don't need to do this in this PR. Re CI slowness, I don't think checking the links will take much CI time, do you?